apps/chef_objects/src/chef_key_base.erl

```diff
@@ -164,3 +165,27 @@ value_or_undefined(Key, Data) ->
       Value
   end.
 
+% Will ensure that if required, only one of 'create_key' or 'public_key' is present,
+% and that the one present is valid.
+-spec validate_public_key_fields(opt|req,  ej:json_object()) -> {ok, ej:json_object()}. % or throw
+validate_public_key_fields(Req, EJ) ->
+    % Simple case first - if one or both are provided and are not valid, handle it.
+    chef_object_base:validate_ejson(EJ, {[ {{opt, <<"create_key">>}, boolean},
+                                           chef_key_base:public_key_spec(opt) ]}),
+    % Now we know that if present, they're good (validate_ejson will throw)
+    case {Req, ej:get({<<"public_key">>}, EJ), ej:get({<<"create_key">>}, EJ)} of
+        {opt, undefined, undefined} ->
```

Comment:
Yeah, I was thinking that `{_, PK, Gen} would match in that case, but the guard would prevent it.  I will correct pre-merge.